Arrowgene.Ddon.Database/Sql/Core/Migration/00000004_UIDMigration.cs

```diff
@@ -0,0 +1,197 @@
+            );
+
+            // item_uid, character_id, storage_type, slot_no, item_num
+            List<(string, uint, ushort, ushort, uint)> ddon_storage_items = new List<(string, uint, ushort, ushort, uint)>();
+            db.ExecuteReader(conn, "SELECT * from ddon_storage_item;",
+               command => { },
+               reader =>
+               {
+                   while (reader.Read())
+                   {
+                       var item = (
+                            db.GetString(reader, "item_uid"),
+                            db.GetUInt32(reader, "character_id"),
+                            db.GetUInt16(reader, "storage_type"),
+                            db.GetUInt16(reader, "slot_no"),
+                            db.GetUInt32(reader, "item_num")
+                       );
+                       ddon_storage_items.Add(item);
+                   }
+               });
+
+            // Drop the tables
+            db.Execute(conn, "DROP TABLE ddon_item;");
+            db.Execute(conn, "DROP TABLE ddon_storage_item;");
+            db.Execute(conn, "DROP TABLE ddon_equip_item;");
+            db.Execute(conn, "DROP TABLE ddon_equip_job_item;");
+
+            // Rename new tables
+            db.Execute(conn, "ALTER TABLE ddon_storage_item_temp RENAME TO ddon_storage_item;");
+            db.Execute(conn, "ALTER TABLE ddon_equip_item_temp RENAME TO ddon_equip_item;");
+            db.Execute(conn, "ALTER TABLE ddon_equip_job_item_temp RENAME TO ddon_equip_job_item;");
+
+            // Drop all existing items
+            db.Execute(conn, "DELETE FROM ddon_storage_item;");
+
+            uint uid = 1;
+            foreach (var storageItem in ddon_storage_items)
+            {
+                var item = ddon_items[storageItem.Item1];
+                var newUid = $"{uid:X08}";
+
+                // item_uid, character_id, storage_type, slot_no, item_num
+                db.ExecuteNonQuery(conn, "INSERT INTO ddon_storage_item (item_uid, character_id, storage_type, slot_no, item_id, item_num, unk3, color, plus_value) VALUES(@item_uid, @character_id, @storage_type, @slot_no, @item_id, @item_num, @unk3, @color, @plus_value);",
+                command =>
+                {
+                    db.AddParameter(command, "item_uid", newUid);
+                    db.AddParameter(command, "character_id", storageItem.Item2);
+                    db.AddParameter(command, "storage_type", storageItem.Item3);
+                    db.AddParameter(command, "slot_no", storageItem.Item4);
+                    db.AddParameter(command, "item_id", item.ItemId);
+                    db.AddParameter(command, "item_num", storageItem.Item5);
+                    db.AddParameter(command, "unk3", item.Unk3);
+                    db.AddParameter(command, "color", item.Color);
+                    db.AddParameter(command, "plus_value", item.PlusValue);
+                });
+
+                uid += 1;
+            }
+
+#if false
```

Comment:
What do you need this for? Can't you insert equip items back into the table by inserting the old data but with the new uid?

Reply:
So there is a problem, since the equipment PR, I don't think it is possible to if the equipped items are unique or if they were assigned to different items. So I opted to just drop all the templates. The player logs in with whatever gear they had equipped, but they are effectively naked until they reequip everything.

Reply:
Why would it not be possible? It'd point to only one instance of the item instead of being able to handle any identical copies of the item, but i think it's better than to suddenly have your character naked. You could have it point to the first item with the same old uid

Reply:
Say I have 4 lanterns. For each job, is the lantern in the equip table the same item, or is each one a separate lantern? The answer is both are valid with the changes we made.

Reply:
You have 4 lanterns, all the same UID: ABCD
Now migrate
You have 4 lanterns, UIDs AAAA BBBB CCCC and DDDD
Have ddon_equip_item point to AAAA for all jobs that had the lantern ABCD
Is it ideal? No. Is it better than be naked? I think it would be

Reply:
What about items where you can have duplicates like jewelry?

Reply:
It's still doable by storing a dictionary with oldUID as a key, and a list of new UIDs for that old UID as a value. For every job for the equipment templates you make a copy of it and pop a new UID from the lists, that way if an item is duplicate you'll get a different UID, but it'll always be the same ones in order across jobs

Reply:
My latest commit now handles those situations when possible